File: 📌 ComputerScience/알고리즘_문풀/Python/BOJ/Boj10866.py
import sys

class MyDequeue:
    def __init__(self):
        self.deq = []

    def push_front(self, x):
        self.deq = [x] + self.deq

    def push_back(self, x):
        self.deq += [x]

    def pop_front(self):
        if self.isEmpty():
            print(-1)
        else:
            print(self.deq.pop(0))

    def pop_back(self):
        if self.isEmpty():
            print(-1)
        else:
            print(self.deq.pop(-1))

    def size(self):
        print(len(self.deq))

    def empty(self):
        print(1 if len(self.deq) == 0 else 0)

    def isEmpty(self):
        return len(self.deq) == 0

    def front(self):
        if self.isEmpty():
            print(-1)
        else:
            print(self.deq[0])

    def back(self):
        if self.isEmpty():
            print(-1)
        else:
            print(self.deq[-1])


# main-method
if __name__ == "__main__" :
    n = int(sys.stdin.readline().rstrip())
    deq = MyDequeue()

    for _ in range(n):
        command = sys.stdin.readline().rstrip()

        if "push_front" in command:
            deq.push_front(command.split()[1])
        elif "push_back" in command:
            deq.push_back(command.split()[1])
        elif "pop_front" in command:
            deq.pop_front()
        elif "pop_back" in command:
            deq.pop_back()
        elif "size" in command:
            deq.size()
        elif "empty" in command:
            deq.empty()
        elif "front" in command:
            deq.front()
        elif "back" in command:
            deq.back()

File: 알고리즘/JAVA/BOJ/Boj10866.java
package com.algorithm.boj;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Deque;
import java.util.LinkedList;

public class Boj10866 {

	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		Deque <Integer> deq = new LinkedList<>();
		
		int count = Integer.parseInt(br.readLine());
		
		while(count-- > 0) {
			String command = br.readLine();
			
			if(command.contains("push_front")) {
				int x = Integer.parseInt(command.split(" ")[1]);
				deq.addFirst(x);
			}else if(command.contains("push_back")) {
				int x = Integer.parseInt(command.split(" ")[1]);
				deq.addLast(x);
			}else if(command.contains("pop_front")) {
				System.out.println(deq.isEmpty() ? -1 : deq.pollFirst());			
			}else if(command.contains("pop_back")) {
				System.out.println(deq.isEmpty() ? -1 : deq.pollLast());				
			}else if(command.contains("size")) {
				System.out.println(deq.size());
			}else if(command.contains("empty")) {
				System.out.println(deq.isEmpty() ? 1 : 0);
			}else if(command.contains("front")) {
				System.out.println(deq.isEmpty() ? -1 : deq.getFirst());
			}else if(command.contains("back")) {
				System.out.println(deq.isEmpty() ? -1 : deq.getLast());				
			}
		}
	}

}
